modules/lambda@edge/main.tf

```diff
@@ -0,0 +1,94 @@
+locals {
+  enabled                   = module.this.enabled
+  destruction_delay_enabled = local.enabled && try(length(var.destruction_delay), 0) > 0
+  functions                 = local.enabled ? var.functions : {}
```

Comment:
did you try to plan the module with `local.enabled=false`?
Terraform usually complains that both sides of the `:` operator must have the same types and the same item types.
If TF does that in this case, then the solution can be:

```
functions  = {for k,v in var.functions : k => v if local.enabled }
```

Reply:
@aknysh yes, the test suite has a few cases where `enabled=false`, and there wasn't an issue.